### DISCUSSION TITLE
This predicate does not override another predicate

### DISCUSSION THREAD

**Discussion by Sim4n6** [github/codeql#13089] [Q&A] [answered]
Please, I am unable to use an `isAdditionalTaintStep()` predicate in a Taint Tracking configuration for Javascript, because I am getting this message error `This predicate does not override another predicate`.

I am using the following prototype 👍🏾 

```codeql
class Configuration extends TaintTracking::Configuration {
...
  override predicate isAdditionalTaintStep(
    DataFlow::Node nodeFrom, DataFlow::Node nodeTo, DataFlow::FlowLabel stateFrom,
    DataFlow::FlowLabel stateTo
  ) {
...
}
}
```

What am I doing wrong please?

**Comment by Sim4n6** [comment]
I still have not figure it out , but `isAdditionalTaintStep()` with enabled flow label is missing from TainTracking configuration ?

**Comment by atorralba** [comment, marked as answer]
Hey @Sim4n6! In JavaScript, you can consider the predicate `isAdditionalTaintStep` an override of the basic `isAdditionalFlowStep` from `DataFlow::Configuration`, just with flow labels being fixed as `taint`. Which means that `isAdditionalTaintStep` exclusively handles taint steps.

If you want to add steps using other flow labels, you can override `isAdditionalFlowStep(DataFlow::Node src, DataFlow::Node trg, DataFlow::FlowLabel inlbl, DataFlow::FlowLabel outlbl)` instead.
